### PR TITLE
Limelight3

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/Robot.java [hunk 1]
```diff
@@ -24,8 +25,8 @@ public void robotInit() {
 		initSubsystems();
         LiveWindow.disableAllTelemetry();
         Battery.getInstance().setDefaultCommand(new BatteryDisabler());
-        Limelight.getInstance();
-        initPortForwarding();
+        Photonvision.getInstance();
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/Robot.java:28]: hmmmm

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/AngPIDSupplier.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package edu.greenblitz.tobyDetermined.commands.swerve;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/AngPIDSupplier.java:1]: is this file just refactored? I believe we don't actually use this command supplier, right?

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java [hunk 3]
```diff
@@ -29,8 +35,7 @@ public static Dashboard getInstance() {
 	public static void init() {
 		instance = new Dashboard();
 	}
-
-	private Dashboard() {
+private Dashboard() {
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java:38]: why?

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java [hunk 4]
```diff
@@ -1,83 +1,66 @@
...
+        return instance;
+    }
+
+    public static void init(){
+        instance = new Limelight();
+    }
+
+
+    public Optional<Pair<Pose2d, Double>> getUpdatedPoseEstimator() {
+        double[] poseArray = robotPoseEntry.getDoubleArray(new double[7]);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java:45]: magic nums, magic nums everywhere

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/AngPIDSupplier.java (hunk 2 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/AngPIDSupplier.java:1]: yes

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java (hunk 4 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java:46]: added doc do not think magic numbers are avoidable

Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java:46]: at least not the indices

Comment (yonnbro) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java:46]: doc is good enough

**Review by yonnbro — APPROVED** [review]
please wait for @roygold8 comments before merging

**Review by roygold8 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java [hunk 5]
```diff
@@ -253,16 +257,31 @@ public void rotateModuleByPower(Module module, double power) {
 		getModule(module).setRotPowerOnlyForCalibrations(power);
 	}
 	
-	public void updatePoseEstimation() {
+	public void updatePoseEstimationPhotonVision() {
```
[the remark this answers is not in the document]
Comment (yonatan-orca) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java:260]: Yeah, sure that works too

**Review by yonnbro — APPROVED** [review]

**Pull request merged**